### PR TITLE
Implement origin business for ILinkFinder contract

### LINKED ISSUE #10: Steam Linker Business implementation
In UI, user will have two buttons:

**Search:**

- Taking into account the detected platforms (Except Steam in the MVP), is going to search in each of one those, the different applications potentially linkable to Steam platform

**Link:**
A selectable grid will add the different applications back create a link in Steam platform.
 
Notes: Take into account threads to improve the implementation

### PR DESCRIPTION
**Pull request by vvillena**
#10 
- Add new persistance layer
- Add repository to DI
- Implementation of Origin platform business (ILinkFinder contract)

### CODE REVIEW

File: Chustasoft.GamersPlatformUtils.Infrastructure/Contracts/IDirectoryRepository.cs [hunk 1]
```diff
@@ -0,0 +1,15 @@
+﻿using System.Collections.Generic;
+using System.IO;
+
+namespace ChustaSoft.GamersPlatformUtils.Infrastructure
+{
+    public interface IDirectoryRepository
```
Comment (xaberue) [inline Chustasoft.GamersPlatformUtils.Infrastructure/Contracts/IDirectoryRepository.cs:6]: Repo name indicates where actions are going to be applied. Suggestions for method names:
- Clean
- Write
- Read

File: ChustaSoft.GamersPlatformUtils.Domain.Origin/Implementations/OriginBusiness.cs [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        private const string LIBRARY_XML_FIRST_LEVEL_TAG = "Settings";
+
+        public bool Available => IsAvailable();
+        public string Name => "Origin";
+        public string Brand => "Electronic Arts";
+        public string AppPath { private set; get; }
+        public string ConfigXMLPath { private set; get; }
+        public IEnumerable<string> Libraries => throw new NotImplementedException();
+
+        private readonly IFileRepository FileRepository;
```
Comment (xaberue) [inline ChustaSoft.GamersPlatformUtils.Domain.Origin/Implementations/OriginBusiness.cs]: Field name -> _fileRepository

**Review by Arthus15 — DISMISSED** [review]

**Review by xaberue — CHANGES_REQUESTED** [review]

File: Chustasoft.GamersPlatformUtils.Infrastructure/Contracts/IDirectoryRepository.cs (hunk 1 above)
Comment (vvillena) [reply Chustasoft.GamersPlatformUtils.Infrastructure/Contracts/IDirectoryRepository.cs:6]: Naming fixed

**Review by xaberue — APPROVED** [review]
Approved!

**Pull request merged**